NEWS

```diff
@@ -9,9 +9,27 @@ Zeek 6.1.0
 Breaking Changes
 ----------------
 
+* ``assert`` is now a reserved keyword for the new ``assert`` statement.
+
 New Functionality
 -----------------
 
+- Added a new ``assert`` statement for assertion based testing and asserting
+  runtime state.
+
+	assert <expr: bool>[, <message: string>];
+
+  This statement comes with two hooks. First, ``assertion_failure()`` that
+  is invoked for every failing assert statement. Second, ``assertion_result()``
+  which is invoked for every assert statement with the resulting outcome and
+  location. The latter allows to construct a summary of failing and passing
+  assert statements. Either hook receives the location and call stack for the
+  ``assert`` statement via a ``Backtrace`` vector.
+
+  A failing assert will abort script execution similar to scripting errors
+  and by default log a reporter error message. Implementing an ``assertion_failure()``
+  hook can be used to suppress the log message for customization.
```

Comment:
One more thought: Now that we don't use the `break` anymore for flagging wether to continue, how about using it for saying whether the normal log message should appear? That would be the default, and if somebody says `break`, the log is suppressed. Does that make sense?

Reply:
> Does that make sense?

Very much - I didn't like the fact "existence of hook handler changes logging behavior" but didn't get to re-using `break` as the signal.

Thanks!